clients/src/main/java/org/apache/kafka/clients/producer/internals/ProduceRequestResult.java

```diff
@@ -32,21 +34,31 @@
     private final CountDownLatch latch = new CountDownLatch(1);
     private volatile TopicPartition topicPartition;
```

Comment:
Not from this patch, but it seems we could make this field `final` if we pass the topic partition into the constructor.